src/dac.rs

```diff
@@ -0,0 +1,179 @@
+            Self::Timer15 => 0b011,
+            Self::Timer2 => 0b100,
+            Self::Timer4 => 0b101,
+            Self::Exti9 => 0b110,
+            Self::SoftwareTrigger => 0b111,
+        }
+    }
+}
+
+pub struct Dac {
+    regs: pac::DAC,
+    channel: Channel,
+    bits: DacBits,
+    vref: f32,
+}
+
+impl Dac {
+    /// Create a new DAC instance
+    pub fn new<P: Pin>(regs: pac::DAC, _pin: P, bits: DacBits, vref: f32) -> Self {
+        Self {
+            regs,
+            channel: P::CHANNEL,
+            bits,
+            vref,
+        }
+    }
+
+    /// Enable the DAC.
+    pub fn enable(&mut self, apb1: &mut APB1) {
+        apb1.enr().modify(|_, w| w.dac1en().enabled());
+        self.regs.cr.modify(|_, w| match self.channel {
+            Channel::One => w.en1().enabled(),
+            Channel::Two => w.en2().enabled(),
+        });
+    }
+
+    /// Disable the DAC
+    pub fn disable(&mut self, apb1: &mut APB1) {
+        self.regs.cr.modify(|_, w| match self.channel {
+            Channel::One => w.en1().disabled(),
+            Channel::Two => w.en2().disabled(),
+        });
+        apb1.enr().modify(|_, w| w.dac1en().disabled());
+    }
+
+    /// Set the DAC value as an integer.
+    pub fn set_value(&mut self, val: u32) {
+        match self.channel {
+            Channel::One => match self.bits {
+                DacBits::EightR => self.regs.dhr8r1.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveL => self.regs.dhr12l1.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveR => self.regs.dhr12r1.modify(|_, w| unsafe { w.bits(val) }),
+            },
+            Channel::Two => match self.bits {
+                DacBits::EightR => self.regs.dhr8r2.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveL => self.regs.dhr12l2.modify(|_, w| unsafe { w.bits(val) }),
+                DacBits::TwelveR => self.regs.dhr12r2.modify(|_, w| unsafe { w.bits(val) }),
+            },
+        }
+    }
+
+    /// Set the DAC voltage.
+    pub fn set_voltage(&mut self, volts: f32) {
+        let val = match self.bits {
+            DacBits::EightR => ((volts / self.vref) * 255.) as u32,
+            DacBits::TwelveL => ((volts / self.vref) * 4_095.) as u32,
+            DacBits::TwelveR => ((volts / self.vref) * 4_095.) as u32,
+        };
+
+        self.set_value(val);
+    }
+
+    // Select and activate a trigger. See f303 Reference manual, section 16.5.4.
+    pub fn set_trigger(&mut self, trigger: Trigger) {
+        self.regs.cr.modify(|_, w| match self.channel {
+            Channel::One => {
+                w.ten1().enabled();
+                unsafe { w.tsel1().bits(trigger.bits()) }
+            }
+            Channel::Two => {
+                w.ten2().enabled();
+                w.tsel2().bits(trigger.bits())
+            }
+        });
+    }
+
+    /// Takes the value stored via set_value or set_voltage and converts it to output on the Pin.
```

Comment:
I did not understand this comment on the first read without context. 

And maybe add
```suggestion
    /// Takes the value stored via set_value or set_voltage and converts it to output on the Pin.
    /// 
    /// Does nothing, if software trigger is not enabled, see `set_trigger()`. 
```